### PR TITLE
breaking: Migrate to beemo v2.

### PR DESCRIPTION
**Pull request by rajzik**
Note:

# @oriflame/lumos was migrated to Beemo v2

[Beemo documentation](https://beemo.dev/docs).
[Beemo migration notes](https://beemo.dev/docs/migration/2.0).

## Breaking changes

- configs are included in lumos
```diff
  "devDependencies": {
-    "@oriflame/config-babel": "4.3.0",
-    "@oriflame/config-eslint": "4.3.0",
-    "@oriflame/config-jest": "4.3.0",
-    "@oriflame/config-prettier": "4.3.0",
-    "@oriflame/config-typescript": "4.3.0",
-     "@oriflame/lumos": "4.3.0",
+     "@oriflame/lumos": "5.0.0",
  }
```
- `lumos-setup` and `lumos-eject` was removed
- Configuration was removed from package.json to `.config/lumos.ts`
```diff
-  "lumos": {
-    "drivers": [
-      "babel",
-      "eslint",
-      "jest",
-      "prettier",
-      "typescript"
-    ],
-    "settings": {
-      "react": true,
-      "testingLibrary": true,
-      "library": true,
-      "future": true,
-      "coverage": 97,
-      "node": true,
-      "buildFolder": "esm"
-    }
-  },
```
- Configs were moved from `config/[driver-name].js` to `.config/lumos/[driver-name].ts`
- Jest no longer need NODE_ENV and TZ env
```diff
-   "jest": "cross-env NODE_ENV=test TZ=UTC lumos jest",
+   "jest": "lumos jest",
```
- typescript `--reference-workspaces` was removed
```diff
-   "build": "lumos typescript --build --reference-workspaces",
+   "build": "lumos typescript --build",
```
- typescript no longer generates tsconfigs in packages automatically. Run `lumos typescript:sync-project-refs` to update them.
```diff
+   "sync-projects": "lumos typescript:sync-project-refs",
```

### CODE REVIEW

**Comment by rajzik** [comment]
TODO List:
[x] Migrate to beemo v2
[ ] Fix issues with running
[ ] Add back tsc build
[ ] Update lumos setup

File: .eslintrc.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-    "./packages/config-eslint/lib/presets/node",
-    "./packages/config-eslint/lib/presets/typescript",
-    "./packages/config-eslint/lib/presets/prettier",
-    "./packages/config-eslint/lib/presets/SECRET_CONFIG_DO_NOT_USE_THIS"
+    "oriflame",
+    "oriflame/node",
+    "oriflame/typescript",
+    "oriflame/future",
+    "oriflame/lib/presets/SECRET_CONFIG_DO_NOT_USE_THIS",
+    "oriflame/prettier",
```
Comment (jakubmazanec) [inline .eslintrc.js:9]: What are these? Paths, packages? Looks confusing.

File: packages/eslint-config/package.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "eslint-config-oriflame",
```
Comment (jakubmazanec) [inline packages/eslint-config/package.json:2]: Why not @oriflame/eslint-config? TBH I'm really confused about all packages names.

File: packages/eslint-config/package.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    "eslint-plugin-promise": "^5.2.0",
+    "eslint-plugin-react": "^7.27.1",
+    "eslint-plugin-react-hooks": "^4.3.0",
+    "eslint-plugin-testing-library": "^5.0.1",
+    "eslint-plugin-unicorn": "^39.0.0"
+  },
+  "devDependencies": {
+    "@types/confusing-browser-globals": "^1.0.0",
+    "@types/node": "^16.11.13",
+    "eslint": "^8.4.1",
```
Comment (jakubmazanec) [inline packages/eslint-config/package.json:63]: Did you test it, so it won't break like last time when we updated ESLint 7->8? And are the rules and plugins' rules updated?

File: packages/tsconfig/tsconfig.json [hunk 4]
```diff
@@ -1,29 +1,28 @@
 {
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "display": "Oriflame",
+
   "compilerOptions": {
+    "allowJs": false,
```
Comment (jakubmazanec) [inline packages/tsconfig/tsconfig.json:6]: This is default, so can be removed.

File: packages/tsconfig/tsconfig.json [hunk 5]
```diff
@@ -1,29 +1,28 @@
...
     "allowSyntheticDefaultImports": true,
-    "declaration": true,
+    "declaration": false,
```
Comment (jakubmazanec) [inline packages/tsconfig/tsconfig.json:8]: This is default, so can be removed.

File: packages/tsconfig/tsconfig.workspaces.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "display": "Lumos with workspaces",
+
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "composite": true,
+    "declaration": true,
```
Comment (jakubmazanec) [inline packages/tsconfig/tsconfig.workspaces.json:8]: This is default when `composite` is `true`, so can be removed.

File: package.json [hunk 7]
```diff
@@ -32,61 +35,28 @@
...
+    "prettier": "^2.5.1",
+    "rimraf": "^3.0.2",
+    "typescript": "^4.5.4",
+    "webpack": "^5.65.0"
   },
   "resolutions": {
-    "caniuse-lite": "1.0.30001274"
-  }
+    "caniuse-lite": "1.0.30001286"
+  },
```
[the remark this answers is not in the document]
Comment (rajzik) [reply package.json:60]: It is updated by renovate bot. And it fixes vunerability.

File: packages/config-typescript/src/index.ts [hunk 8]
```diff
@@ -1,102 +1,65 @@
...
+    compilerOptions.jsx = 'react-jsx';
   }
 
   if (!library) {
-    options.baseUrl = '.';
-    options.paths = {
+    compilerOptions.baseUrl = '.';
+    compilerOptions.paths = {
       [`${ALIAS_PATTERN}/*`]: [`./${srcFolder}/*`],
     };
```
[the remark this answers is not in the document]
Comment (rajzik) [reply packages/config-typescript/src/index.ts:51]: It is configured in other tools from lumos. It is only available for applications.

File: packages/eslint-config/package.json (hunk 2 above)
Comment (rajzik) [reply packages/eslint-config/package.json:2]: Due to resolution of the eslint configs this is correct name. 

As eslint resolves configs by `eslint-config-*` we can then use `oriflame` in eslintrc.

File: packages/eslint-config/package.json (hunk 3 above)
Comment (rajzik) [reply packages/eslint-config/package.json:63]: It is working, ori-ui and spa starter works. even this repo and tests.

File: .eslintrc.js (hunk 1 above)
Comment (rajzik) [reply .eslintrc.js:9]: See resolution of the eslint. This is files that are in npm package.

File: packages/eslint-config/package.json (hunk 2 above)
Comment (rajzik) [reply packages/eslint-config/package.json:2]: By doing it this way we can create oriflame/node and we don't have to write full package name.
example: 
```
extends: oriflame/node
extends: @oriflame/config-eslint/node
```

File: .eslintrc.js (hunk 1 above)
Comment (jakubmazanec) [reply .eslintrc.js:9]: My bad, I forgot about this ESLint feature.

**Review by jakubmazanec — APPROVED** [review]

**Pull request merged**